tensorflow/core/BUILD

```diff
@@ -2229,7 +2229,9 @@ tf_cuda_library(
             "**/*main.cc",
         ],
     ),
-    hdrs = [":framework_internal_public_headers"],
+    hdrs = [":framework_internal_public_headers",
+            ":core_cpu_base_headers",
```

Comment:
Why is this dependency needed?

Reply:
op_kernel.cc inside framework_internal_impl module will depend on device_factory.h, I changed it to this header file only instead of core_cpu_base_headers,

Reply:
`op_kernel.cc` doesn't seem to be a part of this PR. Can you make the change in the PR that adds the dependency? Seeing how it is used would also make it easier to comment how to change it.

Reply:
@annarev op_kernel.cc is included in [framework_internal_impl_srcs](https://github.com/tensorflow/tensorflow/blob/89c2f7f062ee125bd3783035b1be07b5c0f9ea73/tensorflow/core/framework/BUILD#L243) filegroup, which will be depended by [framework_internal_impl](https://github.com/tensorflow/tensorflow/blob/89c2f7f062ee125bd3783035b1be07b5c0f9ea73/tensorflow/core/BUILD#L2000), op_kernel.cc will use device factory to support subdevice, so this dependence is needed,
do you prefer to move device_factory.h header to framework_internal_public_headers?

Reply:
I guess this is kind of a cycle: framework --> op_kernel --> device_factory --> device --> framework
Let me think a bit about the best way to restructure code in this case.

Reply:
Sorry for delay. I moved device.cc/.h and device_factory.cc/.h to tensorflow/core/framework now: https://github.com/tensorflow/tensorflow/commit/5c455a1960745a4c35c16a6aed7fa6d66eeaff23

Can you check if you can remove the header dependency now?

Reply:
thanks for the update, it works . I have remove the header dependency.